### PR TITLE
Started adding a parser -- very raw, can parse return and basic function defn

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

File: bagel/ast.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class Module(object):
+    def __init__(self, declerations):
```
Comment (nelhage) [inline bagel/ast.py]: [nit] spelling: "declarations"

File: bagel/ast.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+class Function(object):
+    def __init__(self, name, arguments, return_type, body):
+        self._name = name
+        self._arguments = arguments
+        self._return_type = return_type
+        self._body = body
+
+    def __eq__(self, other):
```
Comment (nelhage) [inline bagel/ast.py:16]: Is this really what you're going to want? It seems weird to me to want to compare function AST nodes by anything other than identity, or maybe fully-qualified name.

File: bagel/parser.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        "DEF", "RETURN",
+
+        "NAME", "INTEGER",
+    ])
+
+    @_pg.production("program : module")
+    def program_module(self, p):
+        return p[0]
+
+    @_pg.production("module : declerations")
```
Comment (nelhage) [inline bagel/parser.py]: spelling as above

File: bagel/ast.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+class Return(object):
+    def __init__(self, value):
+        self._value = value
+
+    def __eq__(self, other):
+        return self._value == other._value
+
+
+class Suite(object):
```
Comment (nelhage) [inline bagel/ast.py:33]: ooc is "Suite" standard jargon in some context that I'm just not familiar with? I don't recognize it.

File: bagel/ast.py (hunk 2 above)
Comment (alex) [inline bagel/ast.py:16]: It's pretty useful for testing; the alternative is having a test specific external visitor for comparing them.

File: bagel/ast.py (hunk 4 above)
Comment (alex) [inline bagel/ast.py:33]: I picked it up from the Python grammar a while back; the ruby grammar doesn't really have an equivilant, not sure about other grammars.

File: bagel/ast.py (hunk 2 above)
Comment (nelhage) [inline bagel/ast.py:16]: Hm, fair enough. I suspect you'll want to specialize this to the tests eventually.

**Pull request merged**